docs/source/app-dev/code-snippets/Types.daml

```diff
@@ -7,26 +7,30 @@ module Types where
 import Daml.Script
 import DA.Date
 
-data MyProductType = MyProductType {
-  intField: Int;
-  textField: Text;
-  decimalField: Decimal;
-  boolField: Bool;
-  partyField: Party;
-  timeField: Time;
-  listField: [Int];
-  contractIdField: ContractId SomeTemplate
-}
-
-data MySumType = MySumConstructor1 Int |
-                 MySumConstructor2 (Text, Bool)
+-- PRODUCT_TYPE_DEF_BEGIN
+data MyProductType = MyProductType with
```

Comment:
while we're at it, I changed this to use `with` instead of semicolons and braces.

Reply:
you don’t want to adopt spj-style haskell where you use braces & semicolons for everything? :grin: